modules/islandora_wos/islandora_wos.module

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * @file
+ * Web of Science badges
+ */
+
+/**
+ * Implements hook_menu().
+ */
+function islandora_wos_menu() {
+  return array(
+    'admin/islandora/tools/badges/wos' => array(
+      'title' => 'Web of Science',
+      'description' => 'Configure Web of Science integration.',
+      'page callback' => 'drupal_get_form',
+      'page arguments' => array('islandora_wos_admin_form'),
+      'access arguments' => array('administer site configuration'),
+      'type' => MENU_LOCAL_TASK,
+      'file' => 'includes/admin.form.inc',
+    ),
+  );
+}
+
+/**
+ * Implements hook_block_info().
+ */
+function islandora_wos_block_info() {
+  return array(
+    'islandora_wos_badge' => array(
+      'visibility' => BLOCK_VISIBILITY_LISTED,
+      'pages' => 'islandora/object/*',
+      'cache' => DRUPAL_CACHE_PER_PAGE,
+      'info' => t('Islandora Web of Science Badge'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_block_view().
+ */
+function islandora_wos_block_view($delta = '') {
+  module_load_include('inc', 'islandora_badges', 'includes/utilities');
+  $to_render = array();
+
+  if ($delta == 'islandora_wos_badge') {
+    // Get the WoS AMR username and password from the admin config.
+    $wos_username = variable_get('islandora_wos_username', '');
+    $wos_password = variable_get('islandora_wos_password', '');
+    if (!empty($wos_username) && !empty($wos_password)) {
+      $object = menu_get_object('islandora_object', 2);
+      if ($object) {
+        $doi = islandora_badges_get_doi($object);
+        if ($doi) {
+          // Embed Web of Science.
+          // Set API endpoint URL.
+          // @TODO: Add to admin form.
+          $url = "https://ws.isiknowledge.com/cps/xrpc";
+
+          // Store XML Request in string, as I can't send an XML file by POST.
+          $input_xml = '<?xml version="1.0" encoding="UTF-8" ?>
```

Comment:
Would perhaps make more sense as a HEREDOC? The vast majority of Islandora code defines inline XML as HEREDOCs to avoid having to concatenate strings.